Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean

```diff
@@ -624,6 +631,37 @@ lemma nonsingular_add {x₁ x₂ y₁ y₂ : F} (h₁ : W.Nonsingular x₁ y₁)
   nonsingular_neg <| nonsingular_negAdd h₁ h₂ hxy
 #align weierstrass_curve.nonsingular_add WeierstrassCurve.Affine.nonsingular_add
 
+variable {x₁ x₂ : F} (y₁ y₂ : F) (hx : x₁ ≠ x₂)
+
+/-- The formula x(P₁ + P₂) = x(P₁ - P₂) - ψ(P₁)ψ(P₂) / (x(P₂) - x(P₁))²,
+where ψ(x,y) = 2y + a₁x + a₃. -/
+lemma addX_eq_addX_negY_sub :
+    W.addX x₁ x₂ (W.slope x₁ x₂ y₁ y₂) = W.addX x₁ x₂ (W.slope x₁ x₂ y₁ (W.negY x₂ y₂))
+      - (y₁ - W.negY x₁ y₁) * (y₂ - W.negY x₂ y₂) / (x₂ - x₁) ^ 2 := by
+  simp_rw [slope_of_X_ne hx, addX, negY, ← neg_sub x₁, neg_sq]
+  field_simp [sub_ne_zero.mpr hx]
+  ring1
+
+/-- The formula y(P₁)(x(P₂) - x(P₃)) + y(P₂)(x(P₃) - x(P₁)) + y(P₃)(x(P₁) - x(P₂)) = 0,
+assuming that P₁ + P₂ + P₃ = O. -/
+lemma cyclic_sum_Y_mul_X_sub_X :
+    let x₃ := W.addX x₁ x₂ (W.slope x₁ x₂ y₁ y₂)
```

Comment:
```suggestion
    letI x₃ := W.addX x₁ x₂ (W.slope x₁ x₂ y₁ y₂)
```

Reply:
Thanks for the approval! I actually originally started with `letI`, but after @Multramate [showed](https://github.com/leanprover-community/mathlib4/pull/13845#discussion_r1644509949) that the lemmas are still good for rewrites even with `let`, I switched by his request. Do you still prefer `letI` given that `let` works?

Reply:
I still prefer `letI` but no objection if you want to stick with `let`.

Reply:
I'm fine with `letI` (since I don't want to bother Junyan a third time), but I'm still very unconvinced about its use case. Is there a scenario where `letI` works but `let` definitely doesn't? It seems to me they're doing exactly the same thing.